File: C# Basics/Conditional-Statements-Homework/MultiplicationSign/MultiplicationSign.cs
// Write a program that shows the sign (+, - or 0) of the product of three real numbers, without calculating it. Use a sequence of if operators. 

using System;
using System.Globalization;
using System.Threading;

class MultiplicationSign
{
    static void Main()
    {
        Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;

        double firstNumber = double.Parse(Console.ReadLine());
        double secondNumber = double.Parse(Console.ReadLine());
        double thirdNumber = double.Parse(Console.ReadLine());

        if (firstNumber == 0 || secondNumber == 0 || thirdNumber == 0)
        {
            Console.WriteLine(0);
        }
        else if (firstNumber < 0 || secondNumber < 0 || thirdNumber < 0)
        {
            if (firstNumber < 0 && secondNumber < 0 & thirdNumber > 0)
            {
                Console.WriteLine("+");
            }
            else if (firstNumber < 0 && thirdNumber < 0 && secondNumber > 0)
            {
                Console.WriteLine("+");
            }
            else if (secondNumber < 0 && thirdNumber < 0 && firstNumber > 0)
            {
                Console.WriteLine("+");
            }
            else
            {
                Console.WriteLine("-");
            }
        }
        else
        {
            Console.WriteLine("+");
        }
    }
}

File: JavaScript Basics/Conditional statements - Homework/multiplication-sign.js
function solve(args) {
    var firstNumber = +args[0],
        secondNumber = +args[1],
        thirdNumber = +args[2];

    if (firstNumber === 0 || secondNumber === 0 || thirdNumber === 0) {
        return '0';
    } else if (firstNumber > 0 || secondNumber > 0 || thirdNumber > 0) {
        if (firstNumber > 0 && secondNumber > 0 && thirdNumber > 0) {
            return '+';
        } else if (firstNumber > 0 && secondNumber < 0 && thirdNumber < 0) {
            return '+';
        } else if (secondNumber > 0 && firstNumber < 0 && thirdNumber < 0) {
            return '+';
        } else if (thirdNumber > 0 && firstNumber < 0 && secondNumber < 0) {
            return '+';
        } else {
            return '-';
        }
    } else {
        return '-';
    }
}

console.log(solve([1, 1, 1]));
console.log(solve([-1, 1, 1]));
console.log(solve([-1, -1, -1]));
